compiler/src/dotty/tools/dotc/typer/VarianceChecker.scala

```diff
@@ -189,11 +186,10 @@ class VarianceChecker()(implicit ctx: Context) {
     override def traverse(tree: Tree)(implicit ctx: Context) = {
       def sym = tree.symbol
       // No variance check for private/protected[this] methods/values.
-      def skip =
-        !sym.exists ||
-        sym.isAllOf(PrivateLocal) ||
```

Comment:
I found that this change to `skip` alone is sufficient to pass `tests/neg/i7567.scala` - the other changes are not exercised in a test

Reply:
They are: neg/variances.scala will give a different number of errors.

Reply:
Ah yes I see, my mistake